### PR TITLE
Feat/#90 make view for new supplier

### LINKED ISSUE #90: HACER UNA VISTA PARA NUEVO PROVEEDOR ./  MAKE A VIEW FOR A NEW SUPPLIER .
### EXPLANATION

Al seleccionar nuevo proveedor se pasara a la siguiente vista donde se abrirá un espacio para pedir una serie de datos para ingresar a proveedor  el primer dato que pedirá sera el nombre de la empresa con el titulo "Nombre de la empresa " y debajo estará un espacio para poder escribir el nombre de la empresa a ingresar

When you select new supplier you will go to the next view where a space will open to ask for a series of data to enter the supplier, the first data requested will be the name of the company with the title "Name of the company" and below will be a space to write the name of the company to enter.

### SCREENSHOTS

Image

 Crear este titulo y espacio

Image

### USER STORY

Yo como usuario proveedor quiero que al seleccionar "Añadir nuevo proveedor" se muestre una vista  donde  se muestre la opción de " nombre de la empresa" para que pueda  afiliar mi empresa 

As a supplier user, I want to select "Add new supplier" to display a view where the "company name" option is shown so that I can affiliate my company. 

### ACEPTANCE CRITERIA

En esta vista se dará un espacio con el titulo "nombre de la empresa" para agregarla y pueda afiliarme

In this view there will be a space with the title "name of the company" to add it so that I can become a member.

# Referencias

- Por favor terminar en 3 días. Mas información: https://github.com/Azordev/did-admin-panel/issues/192

### PR DESCRIPTION
**Pull request by JRS-Developer**
### Description

Now the admin can create new providers.

I had to add the email input because without it, it wasn't creating the provider.

fixes #90 

#### Screenshots

image

#### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How to test it

If applicable, describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

1. Go to 'Proveedores'
2. Click on 'Crear nuevo proveedor'
3. Try create a new provider.

### Checklist:

- [x] Doesn't break the current code.
- [x] Passes linters and test, also is building.
- [x] Doesn't have spelling or grammatical problems.
- [x] Doesn't have unnecessary comments or debugging code.
- [x] The branch is updated with the last dev branch changes.

### CODE REVIEW

File: views/Providers/Create/index.tsx [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
 
-const CreateProviderLayout: FC<MutableProviderFormProps> = ({ onSubmit, loading }) => (
-  <>
-    <BackHeader to="/proveedores" />
-    <div className={styles.container}>
-      <CreateProviderForm onSubmit={onSubmit} loading={loading} />
-    </div>
-  </>
+const CreateProviderLayout: FC<MutableProviderUserFormProps> = ({ onSubmit, loading }) => (
+  <div className="flex flex-col hhfull">
```
Comment (Israel-Laguan) [inline views/Providers/Create/index.tsx:9]: Is the class `hhfull` correct?

Comment (JRS-Developer) [reply views/Providers/Create/index.tsx:9]: Yes. https://github.com/Azordev/did-admin-panel/issues/90#issuecomment-1242260163

File: components/Button/index.tsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ const Button: FC<ButtonProps> = ({
 }) => {
   return (
     <button type={type} className={`${className} ${styles.button}`} onClick={onClick}>
-      {withIcon && (
+      {withIcon && iconName && (
```
Comment (heliomar-pena) [inline components/Button/index.tsx]: In this case what about removing `withIcon` prop and just using `iconName` as boolean?

```suggestion
      {!!iconName && (
```

File: views/Providers/Create/Form.tsx [hunk 3]
```diff
@@ -1,27 +1,95 @@
...
+        <CustomInput
+          id="date"
+          placeholder="DD/MM/AA"
+          label="Fecha de inicio"
+          register={register}
+          name="createdAt"
+          required
+          type="date"
+        >
+          {errors.createdAt && <span>La fecha de inicio es requerido.</span>}
```
Comment (heliomar-pena) [inline views/Providers/Create/Form.tsx]:
```suggestion
          {errors.createdAt && <span>La fecha de inicio es requerida.</span>}
```

File: views/Providers/Create/index.tsx (hunk 1 above)
Comment (Israel-Laguan) [reply views/Providers/Create/index.tsx:9]: So it's `height: 100%;`, right? hmm we would need to fix that mixin, expected result is `h-full`, thanks for noticing

@Eghinner please create a ticket for it

**Review by Israel-Laguan — APPROVED** [review]

**Review by heliomar-pena — APPROVED** [review]
Tested and is working. Great work! 

imagen

File: components/Button/index.tsx (hunk 2 above)
Comment (JRS-Developer) [reply components/Button/index.tsx]: Done.

**Review by heliomar-pena — APPROVED** [review]
Good work!

File: views/Providers/Create/Form.tsx (hunk 3 above)
Comment (JRS-Developer) [reply views/Providers/Create/Form.tsx]: Done.

Comment (heliomar-pena) [reply views/Providers/Create/Form.tsx]: I don't see the change, can you check if you've pushed it, please?

Comment (JRS-Developer) [reply views/Providers/Create/Form.tsx]: And now?

**Pull request merged**